dvc/remote/gdrive.py

```diff
@@ -38,13 +38,34 @@ def __init__(self, path):
         )
 
 
-def gdrive_retry(func):
-    retry_re = re.compile(r"HttpError (403|500|502|503|504)")
+def append_error_codes(exc):
+    import json
+    from pydrive2.files import ApiRequestError
+
+    assert isinstance(exc, ApiRequestError)
+
+    if hasattr(exc, "errors"):
+        return
+
+    exc.errors = None
 
+    if exc.args:
```

Comment:
is it always HttpError with JSON? should we check the type?

Reply:
HttpError is part of `from apiclient import errors`, and seems we will need to add new `apiclient` deps to DVC.

Reply:
Added there https://github.com/iterative/dvc/pull/3478/commits/91862981811c4813fbada625903abb3eac5fb915

Reply:
We should be able to use it since pydrive depends on it, or am I missing something?

Reply:
it might be a not exactly correct way of doing this though, it's better to be explicit indeed .... that's why I wanted to handle all of this on the pydrive end to keep all these details local to pydrive ... let's migrate this append_error_details to it? right into the ApiRequestError

Reply:
PyDrive2 PR for this https://github.com/iterative/PyDrive2/pull/18

Reply:
so, we don't need the whole append_error_details right? you will the code here?

Reply:
yes, `append_error_details` will be removed completely (already tested locally)